### PR TITLE
Do not attempt to save comments to FITS if Card is missing

### LINKED ISSUE #2738: Unable to save map after aiaprep: CROTA2 not found
Trying to save an AIA submap, from a file downloaded via VSO, results in an error. The following ~~works with~~ does not work with  `sunpy.data.sample.AIA_171_IMAGE`, or with other images that I download:

```
aia = sunpy.map.Map(sunpy.data.sample.AIA_171_IMAGE)
aia1 = aiaprep(aia)
aia1.save("test.fits")
```

The above gives the error:

```
---------------------------------------------------------------------------
KeyError                                  Traceback (most recent call last)
<ipython-input-260-752b35ded2b2> in <module>()
----> 1 aia_sub.save("test.fits")

~/miniconda/lib/python3.6/site-packages/sunpy/map/mapbase.py in save(self, filepath, filetype, **kwargs)
    930         """
    931         io.write_file(filepath, self.data, self.meta, filetype=filetype,
--> 932                       **kwargs)
    933 
    934 # #### Image processing routines #### #

~/miniconda/lib/python3.6/site-packages/sunpy/io/file_tools.py in write_file(fname, data, header, filetype, **kwargs)
    155         for extension, readername in _known_extensions.items():
    156             if fname.endswith(extension):
--> 157                 return _readers[readername].write(fname, data, header, **kwargs)
    158 
    159     else:

~/miniconda/lib/python3.6/site-packages/sunpy/io/fits.py in write(fname, data, header, **kwargs)
    197     if isinstance(key_comments, dict):
    198         for k, v in key_comments.items():
--> 199             fits_header.comments[k] = v
    200     elif key_comments:
    201         raise TypeError("KEYCOMMENTS must be a dictionary")

~/miniconda/lib/python3.6/site-packages/astropy/io/fits/header.py in __setitem__(self, item, comment)
   2030         # In this case, key/index errors should be raised; don't update
   2031         # comments of nonexistent cards
-> 2032         idx = self._header._cardindex(item)
   2033         value = self._header[idx]
   2034         self._header[idx] = (value, comment)

~/miniconda/lib/python3.6/site-packages/astropy/io/fits/header.py in _cardindex(self, key)
   1652 
   1653         if not indices:
-> 1654             raise KeyError("Keyword {!r} not found.".format(keyword))
   1655 
   1656         try:

KeyError: "Keyword 'CROTA2' not found."
```

### PR DESCRIPTION
**Pull request by Cadair**
This was happening when a keyword was deleted from the header without deleting the corresponding comment.

Fixes #2738

### CODE REVIEW

File: sunpy/io/tests/test_fits.py [hunk 1]
```diff
@@ -81,3 +81,18 @@ def test_extract_waveunit_wavelnthcomment_parentheses():
...
+    data, header = sunpy.io.fits.read(AIA_171_IMAGE)[0]
+    outfile = tmpdir / "test.fits"
+    sunpy.io.fits.write(str(outfile), data, header)
+    assert outfile.exists()
+
+
+def test_extra_comment_write(tmpdir):
+    data, header = sunpy.io.fits.read(AIA_171_IMAGE)[0]
+    header["KEYCOMMENTS"]["TEST"] = "Hello world"
+    outfile = tmpdir / "test.fits"
```
[the remark this answers is not in the document]
Comment (Cadair) [reply sunpy/io/tests/test_fits.py]: The tmpdir is a `pathlib.Path` like object.

File: changelog/2748.bugfix.rst [hunk 2]
```diff
@@ -0,0 +1 @@
+Do not attempt to save a FITS header comment for a keyword which is not in the header. This prevents an error on saving some maps after the metadata had been modified but not the comments.
```
Comment (tiagopereira) [inline changelog/2748.bugfix.rst]: Here it should be "for a keyword **that** is not in the header".

File: sunpy/io/tests/test_fits.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (SolarDrew) [reply sunpy/io/tests/test_fits.py]: I wasn't aware (or had forgotten) you could do this but it's the best thing and I love it. Long live `pathlib.Path`.

File: changelog/2748.bugfix.rst (hunk 2 above)
Comment (SolarDrew) [reply changelog/2748.bugfix.rst]: I was about to disagree, but it [appears you're right](https://en.oxforddictionaries.com/usage/that-or-which). Note though:

> in British English, the word *which* is often used interchangeably with the restrictive *that* ... [this] is not strictly incorrect in American English, but it is generally avoided, especially in formal writing

**Pull request merged**